components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1033,6 +1036,8 @@ where
                     }
                 };
                 self.ctx.snap_mgr.delete_snapshot(&key, a.as_ref(), false);
+            } else if self.fsm.peer.pending_remove {
+                fail_point!("pending_remove_is_true");
```

Comment:
The case is tricky. It checks whether `pending_remove` is checked rather than checking whether it works well if `pending_remove` is skipped. In my opinion, whether `pending_remove` is checked is not important, it's just an implementation details. What the case should check is if destroying a peer is delayed for quite a long time, raftstore can still work as usual.

Reply:
Well, this is unit-test/integration test and check some implementation details is normal--otherwise how to differentiate if the snapshot file is deleted by peer's gc or store's gc. Using other fail_points to disable/enable store gc are also implementation details. 

From user's perspective,  it checked if snapshot files are finally deleted and peer will be destroyed and re-created successfully. This is what user cares.   And the check of is "pending_remove_is_true" is kind of white-box test that snapshot file is not deleted by peer's snap gc. 
The whole change's goal is just to prevent snapshot files being deleted when peer is being destroyed and after it's destroyed, they should be deleted. 
Regarding destroying a peer is delayed or not,  and if raftstore still work, it's still within existing code's 100ms range and covered by existing code.

Reply:
> otherwise how to differentiate if the snapshot file is deleted by peer's gc or store's gc

There is no difference. So when gc is tested, it should check whether files will be deleted at the end instead of checking whether it's deleted in peer gc or store gc.

> Using other fail_points to disable/enable store gc are also implementation details.

I'm not sure what's you refer to. But disable/enable gc is more like a configuration change to me.

> The whole change's goal is just to prevent snapshot files being deleted when peer is being destroyed and after it's destroyed, they should be deleted.

Yes, so it should check the consequence of deleting snapshots while destroying a peer. For example, if destroy peer is skipped for a given time, the snapshot files should not be deleted; or if a peer is being destroy, gc has happened before destroy state is persisted, the node should be able to be restarted and resumed.

Utilizing implementation details for timing is OK, but correctness depends on the details is bad. The goal of writing a unit test is not just to ensure current implementation is correct, it also checks refactoring is correct. So correctness should depend on the behavior instead of whether a field is used or not.